src/slurm_plugin/instance_manager.py

```diff
@@ -466,9 +466,15 @@ def terminate_all_compute_nodes(self, terminate_batch_size):
             logger.error("Failed when terminating compute fleet with error %s", e)
             return False
 
-    def _update_failed_nodes(self, nodeset, error_code="Exception"):
+    def _update_failed_nodes(self, nodeset, error_code="Exception", override=True):
         """Update failed nodes dict with error code as key and nodeset value."""
-        self.failed_nodes[error_code] = self.failed_nodes.get(error_code, set()).union(nodeset)
+        if not override:
+            # Remove nodes already present in any failed_nodes key so to not override the error_code if already set
```

Comment:
Why do we need to remove the nodes from all `error_codes`?

Reply:
in order to not override the error_code if it is already set for those nodes